linkatos.py

```diff
@@ -1,79 +1,131 @@
-                     output['channel']
-            elif output and \
-                    'text' in output and \
-                    prog.search(output['text']) != None and \
-                    output['user'] !=  BOT_ID :
-                response = "It looks like you posted a link. /n \
-                    The link is: " + prog.search(output['text']).group(0)
-                bot_message(output['channel'], response)
-                # return text after the @ mention, whitespace removed
-                return output['text'].strip().lower(), output['channel']
-    return None, None
-
-
-if __name__ == "__main__" :
+    print(output_list)  # print the list of outputs to get them on screen
+
+    if output_list and len(output_list) > 0:
+        for output in output_list:
+            # when there is a message then get the channel
+            channel = output['channel']
+
+            if output and output['text'] and output['user'] != BOT_ID:
+                link = message_contains_a_link(output['text'])
+
+            if link:
+                response = "It looks like you posted a link. \
+                    The link is: " + link.group(0)
+                bot_says(channel,
+                         "It looks like you posted a link. \
+                             The link is: " + link.group(0)
+
+    return (link, channel)
+
+def parse_slack_output(slack_rtm_output):
+    """
+       Not ready ... Verify if there is a 'yes' on the response
+
+    """
+    # default outcome
+    there_is_a_yes = False
+    channel = None
+
+    output_list = slack_rtm_output
+    print(output_list)  # print the list of outputs to get them on screen
+
+    if output_list and len(output_list) > 0:
+        for output in output_list:
+            # when there is a message then get the channel
+            channel = output['channel']
+
+            if output and output['text'] and output['user'] != BOT_ID:
+                there_is_a_yes = (output['text'])
+
+    return yes, channel
+
+
+
+if __name__ == '__main__':
     READ_WEBSOCKET_DELAY = 1 # 1 second delay between reading from firehose
-    if slack_client.rtm_connect() :
+    # verify linkatos connection
+    if slack_client.rtm_connect():
         print("linkatos is connected and running!")
         while True:
-            command, channel = parse_slack_output(slack_client.rtm_read())
-            if command and channel:
-                handle_command(command, channel)
+            print("linkatos is listening")
+
+            # parse the messages. Get 'None' while they're empty
+            link, channel = parse_slack_output(slack_client.rtm_read())
+
+            # handle the command when it is a http address
+            if link and channel:
+                bot_says(channel, "Do you want me to store the link " + \
+                        link + " for you?")
+
+                # parse answerif answer...
+                response = parse_response_from_user(slack_client.rtm_read())
```

Comment:
Tinc curiositat per veure com funciona això de cridar dues vegades `slack_client.rtm_read()` en un mateix loop. Sorpren una mica que funcioni :o Però clar, la documentació és força magre